queue/bytes_queue.go

```diff
@@ -39,17 +40,28 @@ type queueError struct {
 // When verbose flag is set then information about memory allocation are printed
 func NewBytesQueue(initialCapacity int, maxCapacity int, verbose bool) *BytesQueue {
 	return &BytesQueue{
-		array:        make([]byte, initialCapacity),
-		capacity:     initialCapacity,
-		maxCapacity:  maxCapacity,
-		headerBuffer: make([]byte, headerEntrySize),
-		tail:         leftMarginIndex,
-		head:         leftMarginIndex,
-		rightMargin:  leftMarginIndex,
-		verbose:      verbose,
+		array:           make([]byte, initialCapacity),
+		capacity:        initialCapacity,
+		maxCapacity:     maxCapacity,
+		headerBuffer:    make([]byte, headerEntrySize),
+		tail:            leftMarginIndex,
+		head:            leftMarginIndex,
+		rightMargin:     leftMarginIndex,
+		verbose:         verbose,
+		initialCapacity: initialCapacity,
 	}
 }
 
+// Reset removes all entries from queue
+func (q *BytesQueue) Reset() {
+	// Just reset indexes
+	q.tail = leftMarginIndex
+	q.head = leftMarginIndex
+	q.rightMargin = leftMarginIndex
+	q.capacity = q.initialCapacity
```

Comment:
Do we need to set it with reset? Capacity shouldn't be changed because we didn't shrink `q.array`